### ISSUE TITLE
native deps

### ISSUE THREAD

**Issue by joernroeder** [p-mazhnik/flutter-embedding#16]
this is great and got me pretty far on a poc I am working on. what about native dependencies the flutter app uses? usually those get auto linked on build, how would this work in an expo project? Any ideas or breadcrumbs?

**Comment by p-mazhnik** [comment]
`flutter build ios-framework` and `flutter build aar` commands will produce artifacts for the native dependencies as well, so they will become a part of the expo app automatically. 

One exception if there is a shared cocoapods dependency on iOS (npm package depends on A and flutter package also depends on A). In this case, there will be `A.xcframework` produced by `flutter build ios-framework` and `A.xcframework` produced during the build phase for the expo iOS app, leading to build errors about duplicated framework.  
Supporting such case will require an update to iOS scripts, more info can be found here: https://github.com/flutter/flutter/issues/130220
Basically, command `flutter build ios-framework` will be `flutter build ios-framework --no-plugins`, and plugins will be linked directly in Podfile using their podspecs, allowing the Cocoapods version management system to resolve common dependency automatically.

**Comment by joernroeder** [comment]
Interesting bc it sounds like it shouldn't be an issue then. I'll try and isolate my use case bc the example you provided works as expected, but when I tried to built an existing flutter app I already have I got

```
⚠️  ld: Could not find or use auto-linked framework 'CoreAudioTypes': framework 'CoreAudioTypes' not found
❌  Undefined symbols for architecture arm64
┌─ Symbol: _FIRConsentStatusDenied
└─ Referenced from: -[FLTFirebaseAnalyticsPlugin setConsent:withMethodCallResult:] in firebase_analytics[arm64][4](FLTFirebaseAnalyticsPlugin.o)
```

which indicated to me that something didn't get included properly.

**Comment by joernroeder** [comment]
adding this to the pubspec.yml of the expo example breaks it for me.
```
dependencies:
  firebase_core: 3.3.0 # <-- bundling this works fine
  firebase_analytics: 11.2.1 # <-- but adding this fails with the error above.
```

**Comment by p-mazhnik** [comment]
Could be related to https://github.com/firebase/flutterfire/issues/13277

**Comment by p-mazhnik** [comment]
I've re-implemented support for native dependencies on iOS, using solution from my first comment; https://github.com/p-mazhnik/flutter-embedding/pull/17
`firebase_analytics` started to work for me. Could you please take a look and let me know if it works for you as well?

**Comment by joernroeder** [comment]
ohhhhhh, this is 🔥. I'll give this a shot this week and let you know.

**Comment by unlu-tahaaytekin** [comment]
hey @joernroeder, did it work for you?

**Comment by joernroeder** [comment]
Hey @unlu-tahaaytekin yeah, it works in principle. I am still working through some quirks getting my own flutter module/app integrated but the POC works as expected.

**Comment by joernroeder** [comment]
Thank you so much @p-mazhnik 🧡 !
